unable to produce milestones / releases?

our milestone build does no longer work.

@LorenzoBettini @szarnekow any idea?

https://ci.eclipse.org/xtext/job/xtext-monorepo-full-deploy-milestone/

maybe related to the recent tycho 4 update

```
13:44:39  [ERROR] Failed to execute goal org.eclipse.tycho:target-platform-configuration:4.0.7:target-platform (default-target-platform) on project org.eclipse.xtext.tests: Execution default-target-platform of goal org.eclipse.tycho:target-platform-configuration:4.0.7:target-platform failed: Cannot invoke "java.io.File.getName()" because "this.artifactFile" is null -> [Help 1]
```

To run this locally, you can explicitly pass these properties as false (with the -D syntax):

		<maven.deploy.skip>true</maven.deploy.skip>
		<maven.javadoc.skip>true</maven.javadoc.skip>
		<gpg.skip>true</gpg.skip>

I don't understand why we don't have this problem in standard builds, where we already use Tycho 4.0.7;

maybe you could temporarily revert to Tycho 4.0.6?

Maybe we hit a Tycho bug?

The problem

```
[ERROR] Failed to execute goal org.eclipse.tycho:target-platform-configuration:4.0.7:target-platform (default-target-platform) on project org.eclipse.xtext.tests: Execution default-target-platform of goal org.eclipse.tycho:target-platform-configuration:4.0.7:target-platform failed: Cannot invoke "java.io.File.getName()" because "this.artifactFile" is null -> [Help 1]
```

appears in our `org.eclipse.xtext.tests` project.

The target platform configuration is the same and inherited and it works in the other projects...

```14:11:06  [ERROR] Failed to execute goal org.eclipse.tycho:target-platform-configuration:4.0.7:target-platform (default-target-platform) on project org.eclipse.xtext.tests: Execution default-target-platform of goal org.eclipse.tycho:target-platform-configuration:4.0.7:target-platform failed: Cannot invoke "java.io.File.getName()" because "this.artifactFile" is null -> [Help 1]
14:11:06  org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.tycho:target-platform-configuration:4.0.7:target-platform (default-target-platform) on project org.eclipse.xtext.tests: Execution default-target-platform of goal org.eclipse.tycho:target-platform-configuration:4.0.7:target-platform failed: Cannot invoke "java.io.File.getName()" because "this.artifactFile" is null
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
14:11:06      at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
14:11:06      at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
14:11:06      at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
14:11:06      at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
14:11:06      at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
14:11:06      at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
14:11:06      at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
14:11:06      at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
14:11:06      at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
14:11:06      at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
14:11:06      at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
14:11:06      at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
14:11:06      at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
14:11:06      at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
14:11:06      at java.lang.reflect.Method.invoke (Method.java:568)
14:11:06      at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
14:11:06      at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
14:11:06      at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
14:11:06      at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
14:11:06      at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
14:11:06      at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
14:11:06      at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
14:11:06      at java.lang.reflect.Method.invoke (Method.java:568)
14:11:06      at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:52)
14:11:06      at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:161)
14:11:06      at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:73)
14:11:06  Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-target-platform of goal org.eclipse.tycho:target-platform-configuration:4.0.7:target-platform failed: Cannot invoke "java.io.File.getName()" because "this.artifactFile" is null
14:11:06      at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:133)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
14:11:06      at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
14:11:06      at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
14:11:06      at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
14:11:06      at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
14:11:06      at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
14:11:06      at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
14:11:06      at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
14:11:06      at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
14:11:06      at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
14:11:06      at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
14:11:06      at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
14:11:06      at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
14:11:06      at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
14:11:06      at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
14:11:06      at java.lang.reflect.Method.invoke (Method.java:568)
14:11:06      at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
14:11:06      at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
14:11:06      at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
14:11:06      at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
14:11:06      at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
14:11:06      at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
14:11:06      at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
14:11:06      at java.lang.reflect.Method.invoke (Method.java:568)
14:11:06      at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:52)
14:11:06      at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:161)
14:11:06      at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:73)
14:11:06  Caused by: java.lang.NullPointerException: Cannot invoke "java.io.File.getName()" because "this.artifactFile" is null
14:11:06      at org.eclipse.tycho.p2maven.advices.MavenChecksumAdvice.getArtifactProperties (MavenChecksumAdvice.java:53)
14:11:06      at org.eclipse.tycho.p2maven.InstallableUnitPublisher.applyAdvices (InstallableUnitPublisher.java:98)
14:11:06      at org.eclipse.tycho.p2resolver.PomUnits.lambda$addCollectedUnits$1 (PomUnits.java:96)
14:11:06      at java.lang.Iterable.forEach (Iterable.java:75)
14:11:06      at org.eclipse.tycho.p2resolver.PomInstallableUnitStore.addPomDependencyConsumer (PomInstallableUnitStore.java:255)
14:11:06      at org.eclipse.tycho.p2resolver.PomUnits.addCollectedUnits (PomUnits.java:85)
14:11:06      at org.eclipse.tycho.p2resolver.P2DependencyResolver.resolvePomDependencies (P2DependencyResolver.java:302)
14:11:06      at org.eclipse.tycho.core.resolver.DefaultTargetPlatformService.lambda$computeFinalTargetPlatform$2 (DefaultTargetPlatformService.java:147)
14:11:06      at org.eclipse.tycho.core.osgitools.DefaultReactorProject$LazyValue.get (DefaultReactorProject.java:307)
14:11:06      at org.eclipse.tycho.core.osgitools.DefaultReactorProject.computeContextValue (DefaultReactorProject.java:200)
14:11:06      at org.eclipse.tycho.core.resolver.DefaultTargetPlatformService.computeFinalTargetPlatform (DefaultTargetPlatformService.java:103)
14:11:06      at org.eclipse.tycho.core.resolver.DefaultTargetPlatformService.getTargetPlatform (DefaultTargetPlatformService.java:99)
14:11:06      at org.eclipse.tycho.target.TargetPlatformMojo.execute (TargetPlatformMojo.java:51)
14:11:06      at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
14:11:06      at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
14:11:06      at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
14:11:06      at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
14:11:06      at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
14:11:06      at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
14:11:06      at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
14:11:06      at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
14:11:06      at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
14:11:06      at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
14:11:06      at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
14:11:06      at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
14:11:06      at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
14:11:06      at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
14:11:06      at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
14:11:06      at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
14:11:06      at java.lang.reflect.Method.invoke (Method.java:568)
14:11:06      at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:283)
14:11:06      at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:226)
14:11:06      at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:407)
14:11:06      at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:348)
14:11:06      at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
14:11:06      at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
14:11:06      at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
14:11:06      at java.lang.reflect.Method.invoke (Method.java:568)
14:11:06      at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:52)
14:11:06      at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:161)
14:11:06      at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:73)
14:11:06  [ERROR] 
14:11:06  [ERROR] 
14:11:06  [ERROR] For more information about the errors and possible solutions, please read the following articles:
14:11:06  [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
14:11:06  [ERROR] 
14:11:06  [ERROR] After correcting the problems, you can resume the build with the command
14:11:06  [ERROR]   mvn <args> -rf :org.eclipse.xtext.tests
14:11:06  [DEBUG] Shutting down adapter factory; available factories [file-lock, rwlock-local, semaphore-local, noop]; available name mappers [discriminating, file-gav, file-hgav, file-static, gav, static]
14:11:06  [DEBUG] Shutting down 'file-lock' factory
14:11:06  [DEBUG] Shutting down 'rwlock-local' factory
14:11:06  [DEBUG] Shutting down 'semaphore-local' factory
14:11:06  [DEBUG] Shutting down 'noop' factory
14:11:06  [DEBUG] Cleanup org.eclipse.jgit.util.FS$FileStoreAttributes$$Lambda$1028/0x00007fb814dd3188@2a9be4b9 during JVM shutdown
```

--- a/org.eclipse.xtext.testlanguages/META-INF/MANIFEST.MF
+++ b/org.eclipse.xtext.testlanguages/META-INF/MANIFEST.MF
@@ -8,8 +8,6 @@ Bundle-ActivationPolicy: lazy
 Require-Bundle: org.eclipse.xtext;bundle-version="2.35.0",
  org.eclipse.xtext.xtext.generator;bundle-version="2.35.0";resolution:=optional;x-installation:=greedy,
  org.eclipse.emf.codegen.ecore;bundle-version="2.29.0";resolution:=optional;x-installation:=greedy,
- org.eclipse.emf.mwe.utils;bundle-version="1.10.0";resolution:=optional;x-installation:=greedy,
- org.eclipse.emf.mwe2.launch;bundle-version="2.17.0";resolution:=optional;x-installation:=greedy,
  org.eclipse.emf.ecore;bundle-version="2.26.0",
  org.eclipse.emf.common;bundle-version="2.24.0",
  org.antlr.runtime;bundle-version="[3.2.0,3.2.1)",

--- a/org.eclipse.xtext.testlanguages/META-INF/MANIFEST.MF
+++ b/org.eclipse.xtext.testlanguages/META-INF/MANIFEST.MF
@@ -13,7 +13,8 @@ Require-Bundle: org.eclipse.xtext;bundle-version="2.35.0",
  org.antlr.runtime;bundle-version="[3.2.0,3.2.1)",
  org.eclipse.xtext.xbase.lib;bundle-version="2.35.0",
  org.eclipse.xtend.lib;bundle-version="2.35.0",
- org.eclipse.xtext.testing;bundle-version="2.35.0"
+ org.eclipse.xtext.testing;bundle-version="2.35.0",
+ org.eclipse.xtext.common.types;bundle-version="2.35.0"
 Import-Package: org.apache.log4j;version="1.2.24",
  org.hamcrest.core,
  org.junit;version="4.13.2",

--- a/org.eclipse.xtext.testlanguages/build.properties
+++ b/org.eclipse.xtext.testlanguages/build.properties
@@ -6,4 +6,7 @@ bin.includes = model/generated/,\
 source.. = src-gen,\
            src
 src.includes = about.html
-additional.bundles = org.apache.commons.logging
+additional.bundles = org.eclipse.emf.mwe.utils,\
+                     org.eclipse.emf.mwe2.launch,\
+                     org.eclipse.emf.mwe2.lib,\
+                     org.apache.commons.logging

